### PR TITLE
ACS Chat - Adding factory methods for existing models

### PR DESCRIPTION
**Pull request by nikithauc**
# Contributing to the Azure SDK

Please see our [CONTRIBUTING.md](https://github.com/Azure/azure-sdk-for-net/blob/main/CONTRIBUTING.md) if you are not familiar with contributing to this repository or have questions.

For specific information about pull request etiquette and best practices, see [this section](https://github.com/Azure/azure-sdk-for-net/blob/main/CONTRIBUTING.md#pull-request-etiquette-and-best-practices).

### CODE REVIEW

File: sdk/communication/Azure.Communication.Chat/src/Models/ChatModelFactory.cs [hunk 1]
```diff
@@ -71,5 +71,75 @@ public static ChatThreadItem ChatThreadItem(string id, string topic, DateTimeOff
...
+        /// <summary>
+        /// Initializes a new instance of the <see cref="ChatThreadProperties"/> class.
+        /// </summary>
+        /// <param name="id"> Chat message. </param>
+        /// <param name="topic"> Topic of the message content. </param>
+        /// <param name="createdOn"> Created on date time </param>
+        /// <param name="createdBy"> Created by </param>
+        /// <param name="deletedOn"> Deleted on date time </param>
+        /// <returns>A new <see cref="ChatThreadProperties"/> instance for mocking.</returns>
+        public static ChatThreadProperties ChatThreadProperties(string id, string topic, DateTimeOffset createdOn, CommunicationIdentifier createdBy, DateTimeOffset deletedOn) => new ChatThreadProperties(id,topic,createdOn, createdBy, deletedOn);
```
Comment (DominikMe) [inline sdk/communication/Azure.Communication.Chat/src/Models/ChatModelFactory.cs:94]: nit: there are space missing between commas

File: sdk/communication/Azure.Communication.Chat/src/Models/SendChatMessageOptions.cs [hunk 2]
```diff
@@ -8,6 +8,20 @@ namespace Azure.Communication.Chat
     /// <summary> Options for the chat message. </summary>
     public class SendChatMessageOptions
     {
+        //internal SendChatMessageOptions(string content, ChatMessageType messageType, string senderDisplayName, IDictionary<string, string> metadata)
```
Comment (DominikMe) [inline sdk/communication/Azure.Communication.Chat/src/Models/SendChatMessageOptions.cs:11]: Can we delete this comment?

File: sdk/communication/Azure.Communication.Chat/src/Models/ChatModelFactory.cs [hunk 3]
```diff
@@ -71,5 +71,75 @@ public static ChatThreadItem ChatThreadItem(string id, string topic, DateTimeOff
...
+        public static ChatParticipant ChatParticipant(CommunicationIdentifier user, string displayName, DateTimeOffset? shareHistoryTime) => new ChatParticipant(user, displayName, shareHistoryTime);
+
+        ///// <summary>
+        ///// Initializes a new instance of the <see cref="SendChatMessageOptions"/> class.
+        ///// </summary>
+        ///// <param name="content"> Content for the message </param>
+        ///// <param name="messageType">The message type.</param>
+        ///// <param name="senderDisplayName"> Time from which the chat history is shared with the member. The timestamp is in ISO8601 format: `yyyy-MM-ddTHH:mm:ssZ`.</param>
+        ///// <param name="metadata"> Properties bag for custom attributes to the message in the form of key-value pair. </param>        /// <returns>A new <see cref="SendChatMessageOptions"/> instance for mocking.</returns>
+        //public static SendChatMessageOptions SendChatMessageOptions(string content, ChatMessageType messageType, string senderDisplayName, IDictionary<string, string> metadata) => new SendChatMessageOptions(content, messageType, senderDisplayName, metadata);
```
Comment (DominikMe) [inline sdk/communication/Azure.Communication.Chat/src/Models/ChatModelFactory.cs:120]: I guess this should be deleted?

File: sdk/communication/Azure.Communication.Chat/src/Models/ChatModelFactory.cs [hunk 4]
```diff
@@ -71,5 +71,75 @@ public static ChatThreadItem ChatThreadItem(string id, string topic, DateTimeOff
...
+        /// <param name="id"> A server-generated message id. </param>
+        /// <returns>A new <see cref="SendChatMessageResult"/> instance for mocking.</returns>
+        public static SendChatMessageResult SendChatMessageResult(string id) => new SendChatMessageResult(id);
+
+        ///// <summary>
+        ///// Initializes a new instance of the <see cref="TypingNotificationOptions"/> class.
+        ///// </summary>
+        ///// <param name="senderDisplayName"> The display name of the message sender. This property is used to populate sender name for push notifications. </param>
+        ///// <returns>A new <see cref="TypingNotificationOptions"/> instance for mocking.</returns>
+        //public static TypingNotificationOptions TypingNotificationOptions(string senderDisplayName) => new TypingNotificationOptions { };
```
Comment (DominikMe) [inline sdk/communication/Azure.Communication.Chat/src/Models/ChatModelFactory.cs:134]: Also to be deleted

File: sdk/communication/Azure.Communication.Chat/src/Models/ChatModelFactory.cs [hunk 5]
```diff
@@ -71,5 +71,75 @@ public static ChatThreadItem ChatThreadItem(string id, string topic, DateTimeOff
...
+
+    //    /// <summary>
+    //    /// Initializes a new instance of the <see cref="UpdateChatMessageOptions"/> class.
+    //    /// </summary>
+    //    /// <param name="messageId">The id of the chat message.</param>
+    //    /// <param name="content">Content of a chat message. </param>
+    //    /// <param name="metadata">Properties bag for custom attributes to the message in the form of key-value pair. </param>
+    //    /// <returns>A new <see cref="UpdateChatMessageOptions"/> instance for mocking.</returns>
+    //    public static UpdateChatMessageOptions UpdateChatMessageOptions(string messageId, string content, IDictionary<string, string> metadata) => new UpdateChatMessageOptions(messageId,content,metadata);
```
Comment (DominikMe) [inline sdk/communication/Azure.Communication.Chat/src/Models/ChatModelFactory.cs:143]: Same here

File: sdk/communication/Azure.Communication.Chat/src/Models/UpdateChatMessageOptions.cs [hunk 6]
```diff
@@ -8,6 +8,18 @@ namespace Azure.Communication.Chat
     /// <summary> Options for the chat message. </summary>
     public class UpdateChatMessageOptions
     {
+        //internal UpdateChatMessageOptions(string messageId, string content, IDictionary<string, string> metadata)
```
Comment (DominikMe) [inline sdk/communication/Azure.Communication.Chat/src/Models/UpdateChatMessageOptions.cs:11]: This can be deleted

**Review by DominikMe — COMMENTED** [review]
Happy to see this being added! Some comments

File: sdk/communication/Azure.Communication.Chat/src/Models/ChatModelFactory.cs [hunk 7]
```diff
@@ -71,5 +71,75 @@ public static ChatThreadItem ChatThreadItem(string id, string topic, DateTimeOff
         [EditorBrowsable(EditorBrowsableState.Never)]
         public static ChatMessageReadReceipt ChatMessageReadReceipt(CommunicationIdentifier sender, string chatMessageId, DateTimeOffset readOn)
```
[the remark this answers is not in the document]
Comment (nikithauc) [reply sdk/communication/Azure.Communication.Chat/src/Models/ChatModelFactory.cs:72]: What is the api.cs file and how do I regenerate it?

[the remark this answers is not in the document]
Comment (nikithauc) [reply sdk/communication/Azure.Communication.Chat/src/Models/ChatModelFactory.cs:72]: Please verify the updates and let me know if the regeneration is as expected.

**Review by DominikMe — APPROVED** [review]

File: sdk/communication/Azure.Communication.Chat/src/Models/ChatModelFactory.cs (hunk 7 above)
[the remark this answers is not in the document]
Comment (DominikMe) [reply sdk/communication/Azure.Communication.Chat/src/Models/ChatModelFactory.cs:72]: lgtm!

**Pull request merged**